### ISSUE TITLE
module/ewmh does not detect removed workspaces in openbox

### ISSUE THREAD

**Issue by hrjakobsen** [jaagr/polybar#1491]
## Describe the issue
Removed workspaces in openbox are not removed from the workspace list in the ewmh module.

**Expected behavior:**
When workspaces are removed in the WM, it should also be removed from the list.

**Actual behavior:**

When workspaces are added in the WM, they are added to the list. When they are removed in the WM, they stay in the list. 

## To Reproduce
*A minimal but complete config with which the problem occurs:*
```dosini
[bar/example]
width = 100%
height = 40
offset-x = 0
offset-y = 0
radius = 0
fixed-center = false
underline-size = 0

modules-left = ewmh 
modules-center = 
modules-right = 

[module/ewmh]
type = internal/xworkspaces

pin-workspaces = true

enable-click = true 

enable-scroll = false
```
*List any other steps needed to reproduce the issue besides starting polybar with the config you posted above.*

## Polybar Log
*Post everything polybar outputs to the terminal when you run it and the issue occurs below*
```

```

## Screenshots
Default workspaces:
![1540815283](https://user-images.githubusercontent.com/3295293/47649458-a6c31d80-db7d-11e8-98d6-e6356f9fe196.png)
After adding a workspace
![1540815302](https://user-images.githubusercontent.com/3295293/47649496-be020b00-db7d-11e8-8782-8fafc614cb9c.png)
After removing the added workspace
![1540815318](https://user-images.githubusercontent.com/3295293/47649507-c3f7ec00-db7d-11e8-88be-00b7240deee9.png)

## Environment:
* WM: Openbox
* Output of `polybar -vvv`:
```
polybar 3.1.0-git

Features: +alsa +curl -i3 -mpd +network

X extensions: +randr (+monitors) -render -damage -sync -composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```
## Additional context
It doesn't matter if the workspace is added with ObConf, with the middleclick menu or using `wmctrl -n 5`

**Comment by patrick96** [comment]
Was able to reproduce this.

Looks like this is persistent accross restarts of polybar. Will do some more investigating

**Comment by patrick96** [comment]
Found the culprit.

When new workspaces are added in openbox, it adds them to the `_NET_DESKTOP_NAMES` hint, but they are not removed if the number of workspaces is decreased. The `_NET_NUMBER_OF_DESKTOPS` hint is not considered at all even though it is the authorative hint when it comes to the number of desktops. We will fix this together with #1462
